src/docfx/lib/schema/JsonSchema.cs

```diff
@@ -10,15 +10,51 @@ namespace Microsoft.Docs.Build
 {
     internal class JsonSchema
     {
+        // A core subset of JSON schema
+        //-------------------------------------------
+
+        /// <summary>
+        /// Type of the current value.
+        /// </summary>
         [JsonConverter(typeof(OneOrManyConverter))]
         public JsonSchemaType[] Type { get; set; }
 
+        /// <summary>
+        /// The JSON schema that applies to each property if the currnet value is object.
+        /// </summary>
         public Dictionary<string, JsonSchema> Properties { get; } = new Dictionary<string, JsonSchema>();
 
+        /// <summary>
+        /// The JSON schema that applies to the array items if the current value is array.
+        /// </summary>
         public JsonSchema Items { get; set; }
 
+        /// <summary>
+        /// An array of valid values for the current value.
+        /// </summary>
         public JValue[] Enum { get; set; }
 
+        /// <summary>
+        /// Properties that are required to be present.
+        /// </summary>
         public string[] Required { get; set; } = Array.Empty<string>();
+
+        // JSON schema custom extensions
+        //-------------------------------------------
+
+        /// <summary>
+        /// Alternative name used in output HTML <meta> tag. If not set, the original metadata name is used.
+        /// </summary>
+        public string HtmlMetaName { get; set; }
```

Comment:
`HtmlMetaName` and `HtmlMetaHidden` are not clear for me since the schema is a nested object, will they be always describing root level metadata names or current level names?

Reply:
always root level because metadata is a flat list.

Reply:
This follows the same syntax as json schema `required`

Reply:
`required` is not always in root level. So the `HtmlMetadataName` could contains `xxxx.yyy.zzz` or `xxx/yyy/zzz`?

Reply:
metadata is a flat list, it cannot be nested, so there is no need to access sub component.